### PR TITLE
Binary protocol support

### PR DESCRIPTION
**Pull request by asergeyev**
Opening this changeset for comments. See #38 for discussion on spec.
Example is likely going to be in micro-application, similar to @joewalnes 's vmstats.

### CODE REVIEW

File: libwebsocketd/process_endpoint.go [hunk 1]
```diff
@@ -13,22 +13,22 @@ import (
 )
 
 type ProcessEndpoint struct {
```
Comment (asergeyev) [inline libwebsocketd/process_endpoint.go]: Someday we should go thru and write docstrings to public methods, make this look pretty: http://godoc.org/github.com/joewalnes/websocketd/libwebsocketd

**Comment by joewalnes** [comment]
Nice work!

I'm preoccupied with some a load of things right now. Give me a few days to review.

File: help.go [hunk 2]
```diff
@@ -54,6 +54,11 @@ Options:
   --passenv VAR[,VAR...]         Lists environment variables allowed to be
                                  passed to executed scripts.
 
+  --binary={true,false}          Switches communication to binary, process reads
+                                 send to browser as blobs and all reads from the
+                                 browser are immediately flushed to the process.
+                                 Default: true
```
Comment (joewalnes) [inline help.go]: Default is false

**Comment by joewalnes** [comment]
This looks great! Assuming you're happy with how it works, I'm ok with it going into master.

Can you write up a page on the wiki explaining how it works and giving an example.

**Comment by asergeyev** [comment]
Ok, I will find time to run thru changes once again and fix that mistake in --help (default is already false)

**Comment by asergeyev** [comment]
Ok, let's try it in source code, no binaries with this feature yet.

**Pull request merged**